assets/js/stripe.js

```diff
@@ -791,31 +816,38 @@ jQuery( function( $ ) {
 		/**
 		 * Queries the server for a PaymentIntent.
 		 *
+		 * @param {string} savedCard The ID of the saved card. (Optional)
 		 * @return {Promise} A promise that resolves to a payment intent.
 		 */
-		getIntent: function() {
-			return new Promise( function( resolve, reject ) {
-				if ( wc_stripe_form.paymentIntent ) {
-					return resolve( wc_stripe_form.paymentIntent );
-				}
+		getIntent: function( savedCard ) {
+			savedCard = savedCard || 'new';
```

Comment:
This isn't needed. `savedCard` will be `undefined` on the normal flow. `savedCard: savedCard` on line 835 will work in this case too.